SygicMapsKit/Classes/BrowseMap/SYMKBrowseMapViewController.swift

```diff
@@ -0,0 +1,73 @@
+import SygicMaps
+import SygicUIKit
+
+
+public class SYMKBrowseMapViewController: UIViewController {
+    
+    public var useCompass = true
+    
+    override public func loadView() {
+        view = SYMKBrowserMapView()
+    }
+    
+    override public func viewDidLoad() {
+        super.viewDidLoad()
+        initSygicMapsSDK()
+        
+        if let view = (view as? SYMKBrowserMapView) {
+            view.compass.isHidden = !useCompass
+        }
+    }
+    
+    // MARK: - Private Methods
+    
+    private func sygicSDKInitialized() {
+        SYOnlineSession.shared().onlineMapsEnabled = true
+        (view as! SYMKBrowserMapView).setupMapView()
+        setupViewDelegates()
+    }
+    
+    private func setupViewDelegates() {
+        guard let view = view as? SYMKBrowserMapView, let mapView = view.mapView else { return }
+        view.compass.delegate = self
+        mapView.delegate = self
+    }
+    
+}
+
+extension SYMKBrowseMapViewController: SYMapViewDelegate {
+    
+    public func mapView(_ mapView: SYMapView, didChangeCameraPosition geoCenter: SYGeoCoordinate, zoom: CGFloat, rotation: CGFloat, tilt: CGFloat) {
+        guard let view = view as? SYMKBrowserMapView else { return }
+        
+        if useCompass {
+            view.compass.isHidden = (rotation == 0)
+            view.compass.viewModel = SYUICompassViewModel(course: Double(rotation), autoHide: false)
+        }
+    }
+    
+}
+
+extension SYMKBrowseMapViewController: SYUICompassDelegate {
+    
+    public func compassDidTap(_ compass: SYUICompass) {
+        guard let view = view as? SYMKBrowserMapView, let mapView = view.mapView else { return }
+        
+        let rotationAngle = mapView.rotation < 180.0 ? -mapView.rotation : 360.0 - mapView.rotation
+        mapView.rotateView(rotationAngle, withDuration: 0.2, curve: .decelerate, completion: nil)
+        compass.isHidden = true
```

Comment:
prec, handluje si to sam kompas

Reply:
jj moze byt, ja som si len nebol uplne isty ci taku vec ako autohide, na zaklade skryvania sameho seba podla toho ci som na severe, ci je ok ze si to riesi view uplne sam, je to take diskutabilne trochu ci je to len view logika alebo trochu uz aj biznis logika, android tiez riesi logiku skryvania v samotnom view, takze za mna kludne mozme nehat tiez tak

Reply:
nerozhoduje o tom view, rozhoduje o tom ten kto nastavuje parametre do SYUICompassViewModel, view sa len sprava podla tych parametrov. Ked budeme chcet reusnut compass v dalsich xy controleroch, nemusime do kazdeho copy-pastovat logiku jeho spravania a vytvarat massive view controllery.

Reply:
jo, suhlasim